api/desecapi/serializers.py

```diff
@@ -668,15 +667,25 @@ def to_representation(self, instance: models.AuthenticatedUserAction):
 
     def to_internal_value(self, data):
         data = data.copy()  # avoid side effect from .pop
+
+        # calculate code TTL
+        validity_period = self.context['view'].validity_period
+        try:
+            ttl = validity_period.total_seconds()
+        except AttributeError:
+            ttl = None  # infinite
+
+        # decode from single string
         try:
-            # decode from single string
-            unpacked_data = self._unpack_code(self.context['code'])
+            unpacked_data = self._unpack_code(self.context['code'], ttl=ttl)
         except KeyError:
             raise serializers.ValidationError({'code': ['This field is required.']})
         except ValueError:
-            validity = settings.VALIDITY_PERIOD_VERIFICATION_SIGNATURE
+            validity = self.context['view'].validity_period
+            if validity is None:
+                validity = 'unlimited'
             raise serializers.ValidationError({
-                'code': [f'This code is invalid, most likely because it expired (validity: {validity}).']
+                'code': [f'This code is invalid, possibly because it expired (validity: {validity}).']
```

Comment:
Now this can result in an error message reading `This code is invalid, possibly because it expired (validity: unlimited)`, let's avoid that

Reply:
done